### PR TITLE
Implementing the laser in RZ

### PR DESCRIPTION
**Pull request by dpgrote**
This version is not working yet, producing a seg fault in RedistributeCPU when run with the test case Examples/Physics_applications/laser_acceleration/input.rz.

### CODE REVIEW

**Comment by dpgrote** [comment]
This is ready to be merged. The test cases works, giving similar results to what Warp gives for the same setup.

**Comment by dpgrote** [comment]
I don't know why the indentations were wrong. I would guess that the file had tabs in it with the wrong tab stop and the indentation was messed up when the tabs were earlier removed.

File: Source/Laser/LaserParticleContainer.cpp [hunk 1]
```diff
@@ -52,6 +52,7 @@ LaserParticleContainer::LaserParticleContainer (AmrCore* amr_core, int ispecies,
         pp.get("wavelength", wavelength);
         pp.get("e_max", e_max);
         pp.query("do_continuous_injection", do_continuous_injection);
+        pp.query("min_particles_per_mode", min_particles_per_mode);
```
Comment (MaxThevenet) [inline Source/Laser/LaserParticleContainer.cpp]: Could you please add a doc entry for this parameter (and other RZ parameters, if needed)?

Comment (dpgrote) [reply Source/Laser/LaserParticleContainer.cpp]: Done

**Pull request merged**